src/js/letters/containers/AddressSection.jsx

```diff
@@ -8,26 +8,63 @@ import {
   isDomesticAddress,
   isMilitaryAddress,
   isInternationalAddress,
-  invalidAddressProperty,
   addressUpdateUnavailable
 } from '../utils/helpers.jsx';
-import { updateAddress, saveAddress } from '../actions/letters';
+import { saveAddress } from '../actions/letters';
 import Address from '../components/Address';
+import InvalidAddress from '../components/InvalidAddress';
+import AddressContent from '../components/AddressContent';
 
 export class AddressSection extends React.Component {
   constructor() {
     super();
-    this.state = { isEditingAddress: false };
+    this.state = {
+      hasLoadedAddress: false,
+      isEditingAddress: false,
+      editableAddress: {},
+    };
   }
 
-  handleUpdate = () => {
+  componentWillReceiveProps(nextProps) {
+    if (!this.state.hasLoadedAddress && Object.keys(nextProps.savedAddress).length > 0) {
+      this.setState({ hasLoadedAddress: true, editableAddress: nextProps.savedAddress });
+    }
+  }
+
+  handleSave = () => {
     this.setState({ isEditingAddress: false });
 
-    this.props.saveAddress(this.props.address);
+    this.props.saveAddress(this.state.editableAddress);
+  }
+
+  handleChange = (path, update) => {
+    this.setState(({ editableAddress }) => {
+      // reset state code when user changes address country but don't add or
+      // modify state property otherwise
+      return (path === 'country'
+        ? {
+          editableAddress: {
+            ...editableAddress,
```

Comment:
If I remember right, the object spread operator is pretty slow and this is going to be run (along with the validations for potentially all the fields) on every keystroke, so I'd like to optimize it as best we can. I think we can use `Object.assign()` to do the same thing.